tests/dyn-drv/eval-outputOf.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+source ./common.sh
+
+# Without the dynamic-derivations XP feature, we don't have the builtin.
+nix --experimental-features 'nix-command' eval --impure  --expr \
+    'assert ! (builtins ? outputOf); ""'
+
+# Test with a regular old input-addresed derivation: works without
+# ca-derivations and doesn't create a placeholder but just returns the
+# output path.
+nix --experimental-features 'nix-command dynamic-derivations' eval --impure --expr \
+    'let
+       item = import ../dependencies.nix;
+       a = item.outPath;
+	   b = builtins.outputOf (builtins.unsafeDiscardOutputDependency item.drvPath) "out";
+     in builtins.trace a
+       (builtins.trace b
+         (assert a == b; null))'
+
+# Test with content addressed derivation.
+nix eval --impure --expr \
+    'with (import ./text-hashed-output.nix); let
+       a = hello.outPath;
+	   b = builtins.outputOf (builtins.unsafeDiscardOutputDependency hello.drvPath) "out";
+     in builtins.trace a
+       (builtins.trace b
+         (assert a == b; null))'
+
+# Test with derivation-producing derivation.
+nix eval --impure --expr \
+    'with (import ./text-hashed-output.nix); let
+       a = producingDrv.outPath;
+	   b = builtins.outputOf (builtins.builtins.unsafeDiscardOutputDependency producingDrv.drvPath) "out";
+     in builtins.trace a
+       (builtins.trace b
+         (assert a == b; null))'
+
+# Test with unbuilt output of derivation-producing derivation.
```

Comment:
```suggestion
# Test with unbuilt output of derivation-producing derivation.
# This is similar to the previous test, but instead of checking the property on a constant derivation, we check it on a derivation that's from another derivation's output (outPath).
```